workflow/rules/long_read.smk

```diff
@@ -32,82 +32,60 @@ rule count_fastq_reads:
-        "../envs/porechop.yaml"
+        temp("results/{date}/filtered/nanofilt/{sample}.fastq"),
     log:
-        "logs/{date}/trimmed/porechop/adapter_barcode_trimming/{sample}.log",
-    threads: 8
-    shell:
-        "porechop -i {input} -o {output} -t {threads} -v 1 > {log} 2>&1"
-
-
-rule customize_primer_porechop:
-    input:
-        get_artic_primer,
-    output:
-        temp("results/.indicators/replacement_notice.txt"),
+        "logs/{date}/nanofilt/{sample}.log",
+    params:
+        min_length=config["quality-criteria"]["ont"]["min-length-reads"],
+        min_PHRED=config["quality-criteria"]["ont"]["min-PHRED"],
     conda:
-        "../envs/primechop.yaml"
-    log:
-        "logs/customize_primer_porechop.log",
+        "../envs/nanofilt.yaml"
     shell:
-        "(cp {input} $CONDA_PREFIX/lib/python3.9/site-packages/porechop/adapters.py && "
-        'echo "replaced adpaters in adapter.py file in $CONDA_PREFIX/lib/python3.9/site-packages/porechop/adapters.py with ARTICv3 primers" > {output}) '
-        "2> {log}"
+        "NanoFilt --length {params.min_length} --quality {params.min_PHRED} --maxlength 700 {input} > {output} 2> {log}"
 
 
-# Using a low number of threads (2-4) speed up primer-trimming significantly (>2x), even for large files,
-# presumably due to the much higher number of target-sequences for trimming as compared
-# to barcode+adapter-trimming. However, using only one thread is again very slow.
-rule porechop_primer_trimming:
+rule downsample_and_trim_raw:
     input:
-        fastq_in=(
-            "results/{date}/trimmed/porechop/adapter_barcode_trimming/{sample}.fastq"
-        ),
-        repl_flag="results/.indicators/replacement_notice.txt",
+        primer=get_artic_primer,
+        reads="results/{date}/filtered/nanofilt/{sample}.fastq",
+        ref_genome="resources/genomes/main.fasta",
     output:
-        temp("results/{date}/trimmed/porechop/primer_clipped/{sample}.fastq"),
-    conda:
-        "../envs/primechop.yaml"
+        "results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.fasta",
+        "results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.clip.fasta",
+    params:
+        outdir="results/{date}/norm_trim_raw_reads/{sample}",
     log:
-        "logs/{date}/trimmed/porechop/primer_clipped/{sample}.log",
-    threads: 2
+        "results/{date}/norm_trim_raw_reads/{sample}/notramp.log",
+    conda:
+        "../envs/notramp.yaml"
     shell:
-        "(porechop -i {input.fastq_in} -o {output} --no_split --end_size 35 --extra_end_trim 0 -t {threads} -v 1) > {log} 2>&1"
+        "notramp -a -r {input.reads} -p {input.primer} -g {input.ref_genome} -o {params.outdir}"
 
 
-rule nanofilt:
+# rule medaka_consensus_reference:
+use rule assembly_polishing_ont as medaka_consensus_reference with:
     input:
-        "results/{date}/trimmed/porechop/primer_clipped/{sample}.fastq",
+        # Don´t ever use corrected reads as input for medaka, it is supposed to polish with raw-reads!
```

Comment:
```suggestion
```

Ah didn't know that. Thanks for the pointer. Is it clipping the amplicon primers somehow? Else it would introduce a bias towards the, right?